src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+import { normalizeMovieData } from '../../utils/normalizeMovieData';
+import { MovieData } from '../../types/MovieData';
+
+type Props = {
+  addMovie: React.Dispatch<React.SetStateAction<Movie[]>>
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(
+    null,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const findMovie = async () => {
+    setIsLoading(true);
+    try {
+      const dataFromServer = await getMovie(query);
+
+      if ('Error' in dataFromServer) {
+        setError(true);
+      } else {
+        setMovie(normalizeMovieData(dataFromServer as MovieData));
+      }
+    } catch (err) {
+      setError(true);
+      throw err;
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleOnChangeQuery = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+    if (error) {
+      setError(false);
+    }
+  };
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    findMovie();
+  };
+
+  const handleOnClickAddMovie = () => {
+    addMovie((movies : Movie[]) => {
+      return [...movies, movie as Movie];
```

Comment:
You should check if the movie is already in the list (by comparing their `imdbId`) before adding it to avoid duplicates. You can use the `some` method to check if the movie exists in the list. 🎬

Reply:
Thanks, corrected.